client/tests/test_train_test_split.py

```diff
@@ -0,0 +1,237 @@
+
+                if self.test_rows > self.num_rows:
+                    iterator_done = True
+            yield response(value.request_type, iterator_done)
+
+
+class MockStream:
+    name = None
+    version = None
+    model = None
+    _stub = None
+
+    def __init__(self, name, version, num_rows):
+        self.name = name
+        self.version = version
+        self.num_rows = num_rows
+        self._stub = MockGrpcStub(self.num_rows)
+
+
+@pytest.mark.parametrize(
+    "kwargs,should_fail",
+    [
+        (
+            {
+                "test_size": 0.5,
+                "train_size": 0.5,
+                "shuffle": True,
+                "random_state": None,
+            },
+            False,
+        ),
+        (
+            {
+                "test_size": -0.5,
+                "train_size": 0.5,
+                "shuffle": True,
+                "random_state": None,
+            },
+            True,
+        ),
+        (
+            {
+                "test_size": 0.5,
+                "train_size": -0.5,
+                "shuffle": True,
+                "random_state": None,
+            },
+            True,
+        ),
+        ({"test_size": 0.5, "shuffle": True, "random_state": None}, False),
+        ({"test_size": 0.5, "random_state": None}, False),
+        ({"test_size": 0.5}, False),
+        ({"test_size": 0.5, "random_state": 3}, False),
+        ({"train_size": 0.5, "shuffle": True, "random_state": None}, False),
+        (
+            {
+                "test_size": 0.5,
+                "train_size": 0.1,
+                "shuffle": True,
+                "random_state": None,
+            },
+            True,
+        ),
+    ],
+)
+def test_initialization(kwargs, should_fail):
+    train_test_stream = MockStream("name", "variant", 1)
+    dataset = Dataset(train_test_stream)
+    if should_fail:
+        with pytest.raises(Exception):
+            dataset.train_test_split(**kwargs)
+    else:
+        dataset.train_test_split(**kwargs)
+
+
+def test_train_test_split():
+    train_test_stream = MockStream("name", "variant", 1)
+    dataset = Dataset(train_test_stream)
+    train, test = dataset.train_test_split(
+        test_size=0.5, train_size=0.5, shuffle=True, random_state=None
+    )
+
+    for features, label in train:
+        assert np.array_equal(features, np.array([["f1", 1, False]], dtype="O"))
+        assert np.array_equal(label, np.array(["train"]))
+
+    for features, label in test:
+        assert np.array_equal(features, np.array([["f1", 1, False]], dtype="O"))
+        assert np.array_equal(label, np.array(["test"]))
+
+
+def test_train_test_batch():
+    train_test_stream = MockStream("name", "variant", 10)
+    dataset = Dataset(train_test_stream)
+    train, test = dataset.train_test_split(
+        test_size=0.5, train_size=0.5, shuffle=True, random_state=None, batch_size=5
+    )
+    i = 0
+    for features, label in train:
+        if i > 1:
```

Comment:
why is this here? do you just want to read the first row? if yes, can just break at the end?